### DISCUSSION TITLE
PAG播放掉帧

### DISCUSSION THREAD

**Discussion by koxyoxyo** [Tencent/libpag#1761] [Q&A] [answered]
[QM_PAG_Template_11.zip](https://github.com/Tencent/libpag/files/12837312/QM_PAG_Template_11.zip)

sdk 3.x版本可以流畅播放，升级到4.2.88版本后相同的PAG播放掉帧，ios和android都有此问题。通过PAGViewer工具看了一下，图片的解码耗时占大头

想请教一下，这种情况的原因是啥，可以怎么优化呢？
3.x版本可以流畅运行，而4.2.88版本运行掉帧是否和4.x版本渲染引擎的切换有关？

**Comment by domchen** [comment, marked as answer]
检查一下具体的测试条件是不是完全相同的，要排除无关的干扰，4.x应该渲染性能是全面优于3.x版本的。具体到你的问题，用4.2以上版本的PAGImageView替换PAGView播放就行，可以获得肉眼可见的性能提升内存降低。
